hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseMergeHelper.java

```diff
@@ -54,12 +55,15 @@ public abstract class BaseMergeHelper {
   protected Iterator<GenericRecord> getMergingIterator(HoodieTable<?, ?, ?, ?> table,
                                                        HoodieMergeHandle<?, ?, ?, ?> mergeHandle,
                                                        Path bootstrapFilePath,
-                                                       Iterator<GenericRecord> recordIterator) throws IOException {
+                                                       Iterator<GenericRecord> recordIterator) {
     Configuration bootstrapFileConfig = new Configuration(table.getHadoopConf());
-    HoodieFileReader<GenericRecord> bootstrapReader =
-        HoodieFileReaderFactory.getFileReader(bootstrapFileConfig, bootstrapFilePath);
-    return new MergingIterator<>(recordIterator, bootstrapReader.getRecordIterator(),
-        (inputRecordPair) -> HoodieAvroUtils.stitchRecords(inputRecordPair.getLeft(), inputRecordPair.getRight(), mergeHandle.getWriterSchemaWithMetaFields()));
+    try (HoodieFileReader<GenericRecord> bootstrapReader =
+        HoodieFileReaderFactory.getFileReader(bootstrapFileConfig, bootstrapFilePath)) {
+      return new MergingIterator<>(recordIterator, bootstrapReader.getRecordIterator(),
+          (inputRecordPair) -> HoodieAvroUtils.stitchRecords(inputRecordPair.getLeft(), inputRecordPair.getRight(), mergeHandle.getWriterSchemaWithMetaFields()));
```

Comment:
Is the record iterator still valid if we close the reader already ?

Reply:
Good point! We should pull up the reader. I'll check other such places. Thanks.